lib/features/home/presentation/home_controller.dart

```diff
@@ -88,4 +132,193 @@ class HomeController extends ReloadableController {
+      return success.personalAccount;
+    } else if (success is GetStoredTokenOidcSuccess) {
+      return success.personalAccount;
+    }
+    return null;
+  }
+
+  void _handleLogOutAndSignInNewAccount({
+    required Success authenticationViewStateSuccess,
+    required PersonalAccount personalAccount,
+    required OpenAppDeepLinkData openAppDeepLinkData,
+  }) {
+    if (authenticationViewStateSuccess is GetCredentialViewState) {
+      setDataToInterceptors(
+        baseUrl: authenticationViewStateSuccess.baseUrl.toString(),
+        userName: authenticationViewStateSuccess.userName,
+        password: authenticationViewStateSuccess.password,
+      );
+    } else if (authenticationViewStateSuccess is GetStoredTokenOidcSuccess) {
+      setDataToInterceptors(
+        baseUrl: authenticationViewStateSuccess.baseUrl.toString(),
+        tokenOIDC: authenticationViewStateSuccess.tokenOidc,
+        oidcConfiguration: authenticationViewStateSuccess.oidcConfiguration,
+      );
+    }
+
+    _getSessionActionToLogOut(
+      authenticationViewStateSuccess: authenticationViewStateSuccess,
+      personalAccount: personalAccount,
+      openAppDeepLinkData: openAppDeepLinkData,
+    );
+  }
+
+  Future<void> _getSessionActionToLogOut({
+    required Success authenticationViewStateSuccess,
+    required PersonalAccount personalAccount,
+    required OpenAppDeepLinkData openAppDeepLinkData,
+  }) async {
+    try {
+      final sessionViewState = await getSessionInteractor.execute().last;
+
+      sessionViewState.fold(
+        (failure) => _handleGetSessionFailureToLogOut(authenticationViewStateSuccess),
+        (success) => _handleGetSessionSuccessToLogOut(
+          sessionViewStateSuccess: success,
+          authenticationViewStateSuccess: authenticationViewStateSuccess,
+          personalAccount: personalAccount,
+          openAppDeepLinkData: openAppDeepLinkData,
+        ),
+      );
+    } catch (e) {
+      logError('HomeController::_getSessionActionToLogOut:Exception = $e');
+      _handleGetSessionFailureToLogOut(authenticationViewStateSuccess);
+    }
+  }
+
+  void _handleGetSessionSuccessToLogOut({
+    required Success sessionViewStateSuccess,
+    required Success authenticationViewStateSuccess,
+    required PersonalAccount personalAccount,
+    required OpenAppDeepLinkData openAppDeepLinkData,
+  }) {
+    if (sessionViewStateSuccess is GetSessionSuccess) {
+      logoutToSignInNewAccount(
+        session: sessionViewStateSuccess.session,
+        accountId: personalAccount.accountId!,
+        onFailureCallback: ({exception}) {
+          if (exception is UserCancelledLogoutOIDCFlowException) {
+            _deepLinksManager?.autoSignInViaDeepLink(
+              openAppDeepLinkData: openAppDeepLinkData,
+              onFailureCallback: () => _continueUsingTheApp(authenticationViewStateSuccess),
+              onAutoSignInSuccessCallback: _handleAutoSignInViaDeepLinkSuccess,
+            );
+          } else {
```

Comment:
why after user `cancel` logout, we still process autoSignIn?

Reply:
Currently TWP has a problem with logout. It cannot automatically close the browser when executing endSession so we temporarily check this condition to help the user confirm that they have logged out. Of course if TWP solved the logout problem we wouldn't need it.